src/transformState.js

```diff
@@ -5,7 +5,19 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const oneAction of actions) {
+    if (oneAction.type === 'addProperties') {
+      Object.assign(state, oneAction.extraData);
+    } else if (oneAction.type === 'removeProperties') {
+      for (const oneArray of oneAction.keysToRemove) {
+        delete state[oneArray];
+      }
+    } else if (oneAction.type === 'clear') {
+      for (const oneState in state) {
+        delete state[oneState];
+      }
+    }
+  }
```

Comment:
avoid such nesting `else if` conditions
This code is hard to read and maintain
and better use `switch` to manage such conditions